src/rules/noPromiseAsBooleanRule.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    isBinaryExpression,
+    isConditionalExpression,
+    isDoStatement,
+    isForStatement,
+    isIfStatement,
+    isPrefixUnaryExpression,
+    isWhileStatement,
+} from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-promise-as-boolean",
+        description: "Warns for Promises that are used for boolean conditions.",
+        optionsDescription: Lint.Utils.dedent`
+            A list of 'string' names of any additional classes that should also be treated as Promises.
+            For example, if you are using a class called 'Future' that implements the Thenable interface,
+            you might tell the rule to consider type references with the name 'Future' as valid Promise-like
+            types. Note that this rule doesn't check for type assignability or compatibility; it just checks
+            type reference names.
+        `,
+        options: {
+            type: "list",
+            listType: {
+                type: "array",
+                items: { type: "string" },
+            },
+        },
+        optionExamples: [true, [true, "Thenable"]],
+        rationale: Lint.Utils.dedent`
```

Comment:
Per #3983 it's rather confusing how these rules all work together. In order to get this new rule in without confusing users, you'll want to:

* Export examples from a `noPromiseAsBoolean.examples.ts` file in `src/rules/code-examples/`
* Link to the relevant docs pages for other Promise rules in the description here: see https://github.com/palantir/tslint/issues/3983#issuecomment-433939470

Reply:
Fixed

Reply:
Fixed